### PR TITLE
goods done

### PR DESCRIPTION
**Pull request by buccaneer33**

### CODE REVIEW

File: src/app/core-modules/goods-category/goods-category.component.scss [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        list-style: none;
+        margin: 0.08rem 0;
+        width: 80%;
+    }
+    &__link{
+        width: 100%;
+    }
+}
+
+.pagination-module{
```
Comment (preznichenko) [inline src/app/core-modules/goods-category/goods-category.component.scss]: pagination should be a separated component

File: src/app/core-modules/services/goods-category.service.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import { Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class GoodsCategoryService {
+
+  constructor( private http: HttpClient ) {}
+
+  public getGoodsPage(): Observable<any> {
```
Comment (preznichenko) [inline src/app/core-modules/services/goods-category.service.ts]: do not use <any> type

File: src/app/core-modules/services/goods-category.service.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      const jsonPath = '../../../assets/dev-stubs/goods-category/goods-category-page.stub.json';
```
Comment (preznichenko) [inline src/app/core-modules/services/goods-category.service.ts]: move it to static attribute of class
and U can write it like: "assets/dev/stubs/...."

File: src/assets/dev-stubs/goods-category/goods-cat-list.stub.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+[
+    "Goods cat1",
```
Comment (preznichenko) [inline src/assets/dev-stubs/goods-category/goods-cat-list.stub.json]: wrong. use interface/class declaration

File: src/app/app.component.html [hunk 5]
```diff
@@ -1,20 +1,11 @@
...
-    <h2><a target="_blank" rel="noopener" href="https://angular.io/tutorial">Tour of Heroes</a></h2>
-  </li>
-  <li>
-    <h2><a target="_blank" rel="noopener" href="https://angular.io/cli">CLI Documentation</a></h2>
-  </li>
-  <li>
-    <h2><a target="_blank" rel="noopener" href="https://blog.angular.io/">Angular blog</a></h2>
-  </li>
-</ul>
-
```
Comment (preznichenko) [inline src/app/app.component.html]: this file is already fixed by another PR. big PR's are really hard to review.
It's not a GIT flow

File: src/app/app.component.scss [hunk 6]
```diff
@@ -0,0 +1,39 @@
+.wrapper{
+
+    width: 100%;
+
```
Comment (preznichenko) [inline src/app/app.component.scss]: why do U need so much empty rows?

File: src/app/common-modules/footer/footer.component.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-footer',
+  templateUrl: './footer.component.html',
+  styleUrls: ['./footer.component.scss']
+})
+export class FooterComponent implements OnInit {
```
Comment (preznichenko) [inline src/app/common-modules/footer/footer.component.ts]: same issue.
U already planned to push this file into master

File: src/app/core-modules/goods/common/components/side-column/side-column.component.ts [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+@Component({
+  selector: 'app-side-column',
+  templateUrl: './side-column.component.html',
+  styleUrls: ['./side-column.component.scss']
+})
+export class SideColumnComponent {
+
+    @Input()
+    list: GoodsCategoryList[];
+  constructor() { }
```
Comment (preznichenko) [inline src/app/core-modules/goods/common/components/side-column/side-column.component.ts]: line it up

File: src/app/core-modules/goods/common/interfaces/goods-cat-list.interface.ts [hunk 9]
```diff
@@ -0,0 +1,4 @@
+export interface GoodsCategoryList {
```
Comment (preznichenko) [inline src/app/core-modules/goods/common/interfaces/goods-cat-list.interface.ts]: wrong like NewsCateroryList

File: src/app/core-modules/goods/goods-category/goods-category.component.scss [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+        flex-grow: 10;
+        height: 100%;
+        display: flex;
+        flex-direction: column;
+        flex-wrap: nowrap;
+        align-items: center
+    }
+}
+
+.news-list{
```
Comment (preznichenko) [inline src/app/core-modules/goods/goods-category/goods-category.component.scss]: wtf?
what news-list doing in goods-category?

File: src/app/core-modules/goods/goods-category/goods-category.component.scss [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+        list-style: none;
+        margin: 0.08rem 0;
+        width: 80%;
+    }
+    &__link{
+        width: 100%;
+    }
+}
+
+.pagination-module{
```
Comment (preznichenko) [inline src/app/core-modules/goods/goods-category/goods-category.component.scss]: 1. pagination is not a module
2. pagination styles shouldn't be here
3. U already have pagination in news PR

File: src/app/core-modules/goods/goods-category/goods-category.component.ts [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+import { GoodsCategoryList } from '@app/core-modules/goods/common/interfaces/goods-cat-list.interface';
+
+@Component({
+  selector: 'app-goods-category',
+  templateUrl: './goods-category.component.html',
+  styleUrls: ['./goods-category.component.scss']
+})
+export class GoodsCategoryComponent implements OnInit {
+
+    public goodsCategoryPage: BehaviorSubject<GoodsCategoryPage> = new BehaviorSubject(null);
```
Comment (preznichenko) [inline src/app/core-modules/goods/goods-category/goods-category.component.ts]: I think U don't understand types of Observable

File: src/app/core-modules/goods/services/goods-category.service.ts [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+import { GoodsCategoryPage } from '@app/core-modules/goods/interfaces/goods-cat-page.interface';
+import { GoodsCategoryList } from '@app/core-modules/goods/common/interfaces/goods-cat-list.interface';
+import { GoodsItem } from '@app/core-modules/goods/common/interfaces/goods-item.interface';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class GoodsCategoryService {
+    private folderPath = '../../../assets/dev-stubs/goods-category';
```
Comment (preznichenko) [inline src/app/core-modules/goods/services/goods-category.service.ts]: remove it

Comment (buccaneer33) [reply src/app/core-modules/goods/services/goods-category.service.ts]: fixed.

File: src/app/core-modules/goods/goods-category/goods-category.component.scss (hunk 11 above)
Comment (buccaneer33) [reply src/app/core-modules/goods/goods-category/goods-category.component.scss]: fixed. block name is pagination-block

File: src/app/core-modules/goods/goods-category/goods-category.component.scss (hunk 10 above)
Comment (buccaneer33) [reply src/app/core-modules/goods/goods-category/goods-category.component.scss]: fixed

File: src/app/core-modules/goods/common/components/side-column/side-column.component.ts (hunk 8 above)
Comment (buccaneer33) [reply src/app/core-modules/goods/common/components/side-column/side-column.component.ts]: fixed.

File: src/app/app.component.scss (hunk 6 above)
Comment (buccaneer33) [reply src/app/app.component.scss]: fixed

Comment (buccaneer33) [reply src/app/app.component.scss]: fixed

File: src/app/core-modules/goods/common/components/goods-medium-card/goods-medium-card.component.scss [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+    &__image{
+        width: 100%;
+        height: 100%;
+    }
+    &__title{
+        font-size: 1.5rem;
+        color: #00f;
+        margin: 0.01rem;
+    }
+    &__battom{
```
Comment (preznichenko) [inline src/app/core-modules/goods/common/components/goods-medium-card/goods-medium-card.component.scss]: typo: button

File: src/app/core-modules/goods/common/components/side-column/side-column.component.scss [hunk 15]
```diff
@@ -0,0 +1,12 @@
+.category-list{
+    display: block;
+
+    &__item{
+        list-style: none;
```
Comment (preznichenko) [inline src/app/core-modules/goods/common/components/side-column/side-column.component.scss]: this prop is used for ul/ol tags only

File: src/app/core-modules/goods/common/interfaces/goods-item.interface.ts [hunk 16]
```diff
@@ -0,0 +1,7 @@
+export interface GoodsItem {
+    id: number;
+    title: string;
+    description: string;
+    image: string;
+    category: number;
```
Comment (preznichenko) [inline src/app/core-modules/goods/common/interfaces/goods-item.interface.ts]: this prop is link to GoodCategory, not a number

File: src/app/core-modules/goods/goods-category/goods-category.component.ts [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+    public categoriesList: Observable<GoodsCategory[]>;
+    public goodsList;
+    public totalItems;
+    public currentPage = 1;
+
+  constructor( private goodsService: GoodsCategoryService ) {}
+
+    ngOnInit() {
+        this.categoriesList = this.goodsService.getGoodsCatList();
+        this.goodsCategoryPage = this.goodsService.getGoodsPage().pipe(map(data => {
```
Comment (preznichenko) [inline src/app/core-modules/goods/goods-category/goods-category.component.ts]: use subscribe, to set TotalItems

File: src/app/core-modules/goods/services/goods-category.service.ts [hunk 18]
```diff
@@ -0,0 +1,24 @@
...
+})
+export class GoodsCategoryService {
+    private apiUrl = 'assets/dev-stubs/goods-category';
+    private pageUrl = 'goods-category-page.stub.json';
+    private catListPath = 'goods-cat-list.stub.json';
+
+    constructor( private http: HttpClient ) {}
+
+    public getGoodsPage(): Observable<GoodsCategoryPage> {
+        return this.http.get<GoodsCategoryPage>(this.apiUrl + '/' + this.pageUrl );
```
Comment (preznichenko) [inline src/app/core-modules/goods/services/goods-category.service.ts]: do not use concatenation here, use only one constant

File: src/app/core-modules/goods/services/goods-category.service.ts [hunk 19]
```diff
@@ -0,0 +1,24 @@
...
+    }
+    public getGoodsCatList(): Observable<GoodsCategory[]> {
+        return this.http.get<GoodsCategory[]>(this.apiUrl + '/' + this.catListPath);
```
Comment (preznichenko) [inline src/app/core-modules/goods/services/goods-category.service.ts]: same here

File: src/styles.scss [hunk 20]
```diff
@@ -1 +1,6 @@
-/* You can add global styles to this file, and also import other style files */
+body{
+    margin: 0;
+    padding: 0;
+    font-family: Verdana, Geneva, Tahoma, sans-serif;
+    font-size: 14px;
+}
```
Comment (preznichenko) [inline src/styles.scss]: all files should ends with empty line

**Pull request merged**